modules/document_repository/php/files.class.inc

```diff
@@ -82,9 +82,14 @@ class Files extends \NDB_Page
                         )
                     );
             }
+
```

Comment:
This whole section of code looks questionable to me. Why are we parsing the username out of the url, parsing the filename out of the url, re-concatenating them, and then passing it to the `FilesDownloadHandler`? It looks to me like the handler should just be `$downloadpath` and the `$filename` should just be `$request->getUri()->getPath()` with the endpoint name prefix stripped out.

I suspect that at some point that this was hacked in because the subdirectories weren't working until fixed in #8392 but the FilesDownloadHandler should handle subdirectories properly now that @cmadjar fixed it.

Reply:
im not sure, john saigle had written that bit of code at the same time he created the FilesDownloadHandler

Reply:
i'm just fixing the username here

Reply:
@driusan fixed it to what you suggested. can you re-review?